### PR TITLE
[IE Myriad] Correct destruction order in functional tests with DISABLE_PLUGIN_CACHE  env. variable

### PR DESCRIPTION
**Pull request by nikita-kud**
#-25382
During the research, the problem was found that libusb crush - only consequence of destruction order problem.

### CODE REVIEW

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 namespace LayerTestsUtils {
 
 LayerTestsCommon::LayerTestsCommon() {
-    core = PluginCache::get().ie(targetDevice).get();
+    core = PluginCache::get().ie(targetDevice);
```
Comment (gladilov-gleb) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:10]: Could you please explain that change?
Originally, `core` is a raw pointer by intention, so common test fixture doesn't extend lifetime of a core.

Comment (nikita-kud) [reply inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:10]: @gladilov-gleb , please pay attention to [this line](https://github.com/openvinotoolkit/openvino/blob/b6a05c232ed8536689be5e67d7fc597a4f654fef/inference-engine/tests/ie_test_utils/functional_test_utils/plugin_cache.cpp#L38).
When DISABLE_PLUGIN_CACHE is enabled then PluginCache simply returns new shared pointer. If we store a raw pointer, IECore instance is destroyed and we get wild pointer.

Comment (gladilov-gleb) [reply inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:10]: Thanks for clarification!

Comment (nikita-kud) [reply inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:10]: > so common test fixture doesn't extend lifetime of a core.

In any case, ie_core is stored in a `static instance of PluginCache;` -- this means that the test fixture cannot extend lifetime of a core. No?

**Pull request merged**